### PR TITLE
add autoformatting

### PR DESCRIPTION
**Pull request by kleinschmidt**

### CODE REVIEW

**Review by ericphanson — APPROVED** [review]

File: .github/workflows/Format.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  # because we rely on ReviewDog to trigger failure, there's no reason to run on anything
+  pull_request:
+    paths:
+      - "**/*.jl"
+      - ".github/workflows/FormatCheck.yml"
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
```
Comment (ericphanson) [inline .github/workflows/Format.yml:12]: I've been using this, FWIW
```suggestion
name: Format Check
on:
  push:
    branches:
      - 'main'
      - /^release-.*$/
    tags: ['*']
  pull_request:
    types: [opened, synchronize, reopened, ready_for_review]
concurrency:
  # Skip intermediate builds: always.
  # Cancel intermediate builds: only if it is a pull request build.
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}

```

Comment (kleinschmidt) [reply .github/workflows/Format.yml:12]: right, see the comment ;) 

build won't fail unless reviewdog actually runs, and that only runs on PRs (non-draft in this workflow)

File: .github/workflows/Format.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+jobs:
+  format-check:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        version:
+          - "1"
```
Comment (palday) [inline .github/workflows/Format.yml:19]:
```suggestion
          - "1.9"
```

since you're using a manifest

File: .github/workflows/Format.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v1
+        with:
+          cache-name: "${{ github.workflow }}-${{ github.job }}-${{ matrix.pkg.name }}-${{ matrix.version }}-${{ matrix.os }}-${{ matrix.arch }}"
+          cache-compiled: true
+      - name: Instantiate `format` environment and format
+        run: |
+          julia --project=format -e 'using Pkg; Pkg.instantiate()'
+          julia --project=format 'format/run.jl'
+      - run: git diff
```
Comment (palday) [inline .github/workflows/Format.yml:38]:
```suggestion
```

Comment (kleinschmidt) [reply .github/workflows/Format.yml:38]: debugging, let me be

File: .github/workflows/Format.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      - uses: reviewdog/action-suggester@v1
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.draft == false }}
```
[the remark this answers is not in the document]
Comment (kleinschmidt) [reply .github/workflows/Format.yml:40]: this is intentional

File: .github/workflows/Format.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    paths:
+      - "**/*.jl"
+      - ".github/workflows/FormatCheck.yml"
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: only if it is a pull request build.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+jobs:
+  format-check:
```
Comment (palday) [inline .github/workflows/Format.yml:14]:
```suggestion
  format-check:
  if: ${{ github.event_name == 'pull_request' && github.event.pull_request.draft == false }}
```

File: .github/workflows/Format.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kleinschmidt) [reply .github/workflows/Format.yml:40]: (don't want reviewdog cluttering up draft PRs with noise)

[the remark this answers is not in the document]
Comment (palday) [reply .github/workflows/Format.yml:40]: see where I moved the if block to

File: .github/workflows/Format.yml (hunk 3 above)
Comment (palday) [reply .github/workflows/Format.yml:38]: just trying to make it match a known working example as close as possible 😉

Comment (kleinschmidt) [reply .github/workflows/Format.yml:38]: okay, that's this commit then: https://github.com/beacon-biosignals/EDF.jl/pull/73/commits/b9494bd6514813ba400048b9de3a647a64d7cb69

File: test/runtests.jl [hunk 6]
```diff
@@ -156,7 +167,7 @@ const DATADIR = joinpath(@__DIR__, "data")
     #     for x in signal:
     #         f.write("%s\n" % x)
     # ```
-    mne = map(line->parse(Float32, line), eachline(joinpath(DATADIR, "mne_values.csv")))
+    mne = map(line -> parse(Float32, line), eachline(joinpath(DATADIR, "mne_values.csv")))
     for (a, b) in zip(EDF.decode(signal), mne)
         @test a ≈ b atol=0.01
```
[the remark this answers is not in the document]
Comment (kleinschmidt) [reply test/runtests.jl:172]: no, bad bot.

**Comment by kleinschmidt** [comment]
The issue was that none of the formatting changes were in the diff.  That's an expected feature of reviewdog...

**Pull request merged**